src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -0,0 +1,510 @@
+                           "region_name": ("field", "test")})
+    out = str(invoke.gen())
+
+    filename = "driver-field-test.f90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    # Now get all the variable names and variables listed in ProvideVariable
+    # in extracting code, and make sure the same appear in the driver:
+    # This re extracts the quoted variable name and the variable itself,
+    # e.g. '"a", a'. The ReadVariable function takes exactly the same
+    # parameters.
+    provide = re.compile(r"ProvideVariable\((.*)\)")
+    for line in out.split("\n"):
+        # The current `gen` created double quotes, while using PSyIR
+        # produces single quotes. So convert them first:
+        line = line.replace('"', "'")
+        grp = provide.search(line)
+        if grp:
+            # A special handling is required for output variables:
+            # The output value of 'a' is written as '"a_post", a'. But the
+            # driver needs to read this into a different variable called
+            # 'a_post', so we also need to test if appending 'post''
+            params = grp.groups(0)[0]
+            if "_post', " in params:
+                # It's an output variable, so the driver u
+                assert f"({grp.groups(0)[0]}_post)" in driver
+            else:
+                assert f"({grp.groups(0)[0]})" in driver
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir")
+def test_lfric_driver_field_arrays():
+    '''Test handling of array of fields: they are written in one call to
+    the extraction library, but the library will write each array member
+    as an individual field. The driver needs to read in each individual
+    array member into distinct variables.'''
+
+    _, invoke = get_invoke("8_vector_field_2.f90", API,
+                           dist_mem=False, idx=0)
+
+    extract = LFRicExtractTrans()
+
+    extract.apply(invoke.schedule.children[0],
+                  options={"create_driver": True,
+                           "region_name": ("field", "array")})
+    # The extraction provides the array once, it is the responsibility
+    # of the extraction library to create the individual fields.
+    out = str(invoke.gen())
+    assert "ProvideVariable(\"chi\", chi)" in out
+
+    filename = "driver-field-array.f90"
+    with open(filename, "r", encoding='utf-8') as my_file:
+        driver = my_file.read()
+
+    # Check that the driver reads the three individual fields
+    assert "ReadVariable('chi%1', chi_1)" in driver
+    assert "ReadVariable('chi%2', chi_2)" in driver
+    assert "ReadVariable('chi%3', chi_3)" in driver
+
+
+# ----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir")
+def test_lfric_driver_operator():
+    '''Test handling of array of fields: they are written in one call to
```

Comment:
This docstring needs re-writing to talk about operators :-)

Reply:
Indeed.